docs/layout-of-source-files.rst

```diff
@@ -216,101 +216,562 @@ the code below creates new global symbols ``alias`` and ``symbol2`` which refere
+.. code-block:: solidity
+    :caption: /a/b/c/contract.sol
 
-For example, if you clone ``github.com/ethereum/dapp-bin/`` locally to
-``/usr/local/dapp-bin``, you can use the following in your source file:
+    import "../../../../util.sol";       // Import key: /util.sol
+    import "../../../../../util.sol";    // Import key: /util.sol
+    import "../../../../../../util.sol"; // Import key: /util.sol
+
+.. code-block:: solidity
+    :caption: a/b/c/contract.sol
+
+    import "../../../../util.sol";       // Import key: util.sol
+    import "../../../../../util.sol";    // Import key: util.sol
+    import "../../../../../../util.sol"; // Import key: util.sol
+
+After stripping the leading relative segments, the import path is normalized so that the
+resulting import key does not contain any ``./`` or ``../``:
+
+.. code-block:: solidity
+    :caption: /a/b/c/contract.sol
+
+    import "../../d/e///.././util.sol"; // Import key: /a/e/util.sol
+
+This is quite different from imports relative to base where the ``///.././`` part would remain
+in the import key.
+
+Note that the parent import key is **not** normalized, and the ``./`` and ``../`` segments in it
+have no special meaning:
+
+.. code-block:: solidity
+    :caption: ../lib/math.sol
+
+    import "./util.sol" as util;    // Import key: ../lib/util.sol
+    import "../token.sol" as token; // Import key: ../../token.sol
+
+This may lead to surprising results in corner cases:
+
+.. code-block:: solidity
+    :caption: /a/./b/contract.sol
+
+    import "../c/util.sol";       // Import key: /a/./c/util.sol
+    import "../../c/util.sol";    // Import key: /a/c/util.sol
+    import "../../../c/util.sol"; // Import key: /c/util.sol
+
+.. index:: remapping, import remapping
+.. _import-remapping:
+
+Import Remapping
+~~~~~~~~~~~~~~~~
+
+Base path and relative imports allow you to feely move your project around the filesystem but still
+force yout to keep all files within in a single directory and its subdirectories.
+When using external libraries it is often desirable to keep their files in a separate location.
+To help with that, the compiler provides another mechanism: import remapping.
+
+Remapping allows you to use placeholders as import key prefixes and then have the compiler replace
+them with actual paths.
+For example you can set up a remapping so that everything imported from the virtual directory
+``github.com/ethereum/dapp-bin/library`` would actually be read from your local directory
+``/usr/local/dapp-bin/library``.
+
+The remappings can depend on a context, which allows you to configure packages to import,
+e.g. different versions of a library of the same name.
+
+.. warning::
+
+    Information about used remappings is stored in contract metadata so, while they let you avoid
+    changing the source, they cannot be used to achieve reproducible builds.
+    The metadata hash embedded in the bytecode will not be the same if you perform import remapping.
```

Comment:
I was really surprised that this is the case. I think this defeats the purpose of using remappings in many cases.

Reply:
Remappings were actually not meant to be used with local paths. After all, the last part of a remapping is an entry in the virtual filesystem.

Reply:
Do you mean that they were meant to be used only with Standard JSON and solc-js? Because when using CLI, local paths are always involved.